compiler/src/dmd/lexer.d

```diff
@@ -1556,6 +1556,8 @@ class Lexer
                     if (ndigits != 2 && !utf_isValidDchar(v))
                     {
                         error(loc, "invalid UTF character \\U%08x", v);
+                        if (v >= 0xD800 && v <= 0xDFFF)
+                            errorSupplemental("The code unit is a UTF-16 surrogate, is the escape UTF-16 not a Unicode code point?");
```

Comment:
are we able to determine with certainty if it is not a code point?

Reply:
Yes, if this branch is taken.

UTF-16 surrogates are never valid code points. They are an encoding detail.

I wrote it as a question, because for all we know they could be trying to ``\x`` it, rather than trying to encode ``wchars``.

Reply:
can we differentiate between `\x` and `\U` escapes?

Reply:
Yes but we don't need to.

``\x`` is only digit hex, it can't represent the surrogates which are 4.